src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, null, null);
+        newNode.prev = last;
+        if (last != null) {
+            last.next = newNode;
+        }
+        if (first == null) {
+            first = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> newNode = new Node<>(value, null, null);
+            Node<T> tempPoint = getByIndex(index);
+            newNode.prev = tempPoint.prev;
+            newNode.next = tempPoint;
+            if (tempPoint.prev != null) {
+                tempPoint.prev.next = newNode;
+            }
+            tempPoint.prev = newNode;
+            if (newNode.prev == null) {
+                first = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null) {
+            for (T element : list) {
+                add(element);
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> tempPoint = getByIndex(index);
+        return tempPoint != null ? tempPoint.data : null;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T result = null;
+        Node<T> tempPoint = getByIndex(index);
+        if (tempPoint != null) {
```

Comment:
Throw exception in `getByIndex`, check comment in the top

Reply:
ok